FAQ.md

```diff
@@ -25,40 +25,50 @@ Dragonfly makes it simple and cost-effective to set up, operate, and scale any
 
-## What is dfget
+Supernode will download files and cache them via CDN, more info refers to [The sequence of supernode's CDN functionality](#what-is-the-sequence-of-supernode's-cdn-functionality).
 
-Dfget is the client of Dragonfly used for downloading files. It's similar to using wget.
+The dfget will trigger the cache before the download is complete, saving the downloaded file in a specific location (default is “.small-dragonfly/data”, suffixed with ".server").
 
-At the same time, it also plays the role of peer, which can transfer data between each other in p2p network.
+After, there are some things we need to pay attention to:
 
-## What is dfdaemon
+1. Files being transferred over the P2P network must exist on the supernode. If the file does not exist on the supernode, there is no corresponding taskID, and the corresponding bitmap information, it can not be scheduled, that is the file can be transferred in the P2P network only if there is cache information on the supernode.
+2. The cache on the peees is not coupled to the cache on the Supernode, and each is responsible for the life cycle. Then what if the file on the peer is deleted, please refer to [What if you kill the dfget server process or delete the source files](#what-if-you-kill-the-dfget-server-process-or-delete-the-source-files)
 
-Dfdaemon is a local long-running process which is used for translating images pulling request from container engine. It establishes a proxy between container engine and registry.
+## What is the sequence of supernode's CDN functionality
 
-Dfdaemon filters out layer fetching requests from all requests send by dockerd/pouchd when pulling images, then it uses dfget to downloading these layers.
+When dfget register a task to supernode, supernode will check whether the file to be downloaded has a cache locally.
 
-## What is the sequence of supernode's CDN functionality and P2P distribution
+If the file to be downloaded which has not been cached in supernode yet, supernode will do as the following sequence:
 
-When dfget starts to pull an image which has not been cached in supernode yet, supernode will do as the following sequence:
-
-- First Step: supernode triggers the downloading task:
+- First Step: supernode triggers the downloading task asynchronously:
   - fetch the file/image length;
   - divide the length into pieces;
-  - start to download the file piece by piece;
+  - start to download the file piece by piece and store it locally;
 - Second Step:
   - supernode finishes to download one piece
   - supernode starts the P2P distribution work for the downloaded one piece among peers;
 
-In a word, supernode does not have to wait for all the piece downloadings finished, it can concurrently start piece downloading once one piece downloaded.
+If the file to be downloaded which has been cached in supernode, supernode will back to the source to determine if there is an update with headers `If-None-Match:<eTag>` and `If-Modified-Since:<lastModified>`.
```

Comment:
s/back to the/send an HTTP request to the source file network address to determine.....